app/uk/gov/hmrc/exports/services/mapping/goodsshipment/PreviousDocumentsBuilder.scala

```diff
@@ -72,21 +72,30 @@ class PreviousDocumentsBuilder @Inject()() extends ModifyingBuilder[PreviousDocu
 
   private def isDefined(previousDocumentsData: ConsignmentReferences): Boolean = previousDocumentsData.ducr.nonEmpty
 
-  private def createDucrDocument(ducr: DUCR): GoodsShipment.PreviousDocument = {
+  private def createDucrDocument(ducr: DUCR): GoodsShipment.PreviousDocument = createPrevDoc(ducr.ducr, "DCR")
+
+  def buildThenAdd(mucr: MUCR, goodsShipment: GoodsShipment): Unit =
+    if (mucr.nonEmpty) {
+      goodsShipment.getPreviousDocument.add(createMucrDocument(mucr))
+    }
+
+  private def createMucrDocument(mucr: MUCR): GoodsShipment.PreviousDocument = createPrevDoc(mucr.mucr, "MCR")
+
+  private def createPrevDoc(value: String, typeCodeValue: String): GoodsShipment.PreviousDocument = {
     val previousDocument = new GoodsShipment.PreviousDocument()
 
     val categoryCode = new PreviousDocumentCategoryCodeType()
     categoryCode.setValue("Z")
     previousDocument.setCategoryCode(categoryCode)
 
     val id = new PreviousDocumentIdentificationIDType()
-    id.setValue(ducr.ducr)
+    id.setValue(value)
     previousDocument.setID(id)
 
     previousDocument.setLineNumeric(new java.math.BigDecimal(1))
```

Comment:
Should `LineNumeric` always be '1' ? If there are both DUCR and MUCR, then there are 2 elements with the same value. Should they not be different?

Reply:
I believe this line number relates to a line on the 'document' that you are referencing (so line 1 of this document has the relevant references for this declarations). Each previous document element is independent of each other.

Reply:
Just confirmed this with Liz too :+1:

Reply:
Thanks, I was missing the context.